esp-hal/src/i2c/master/mod.rs

```diff
@@ -106,12 +106,38 @@ pub enum Error {
     InvalidZeroLength,
 }
 
+impl core::error::Error for Error {}
+
+impl core::fmt::Display for Error {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            Error::ExceedingFifo => write!(f, "The transmission exceeded the FIFO size"),
```

Comment:
There are crates that can turn docstrings into Display output (displaydoc or docsplay). Should we try and use them?

Reply:
I guess `thiserror` does support `no_std` now, but I'm fairly indifferent.